smdebug/tensorflow/base_hook.py

```diff
@@ -482,19 +484,21 @@ def set_optimizer_variables(self, optimizer_variables):
         This method helps find the optimizer variables (such as momentum)
         :param optimizer_variables: list of tf.Variables/tf.Tensors/tf.MirroredVariables
         """
-        # TF 2.x doesn't provide gradient/optimizer variable names and values by default.
-        # Skipping set_gradients and set_optimizer_variables for Tf 2.x until there is
+        # TF 2.x provides only symbolic gradient variables that do not provide access to their values.
+        # Skipping set_gradients for Tf 2.x until there is
         # support to pass names and values from TF side.
 
         # From TF 2.2, executing_eagerly_outside_functions() can be used as
         # ops.executing_eagerly_outside_functions() or tf.compat.v1.executing_eagerly_outside_functions().
         # But in TF 2.1, only ops.executing_eagerly_outside_functions() is valid
-        if is_tf_version_2x() and ops.executing_eagerly_outside_functions():
-            return
         # since this is done for each variable at a time for keras, not checking if set already
         self.collection_manager.get(CollectionKeys.OPTIMIZER_VARIABLES).add_for_mode(
-            optimizer_variables, ModeKeys.TRAIN
+            optimizer_variables, self.mode
```

Comment:
why this change?

Reply:
Why is there a need to hardcode `ModeKeys.TRAIN`,  will this not lead to a case when the mode is actually EVAL or PREDICT but we save tensors with `ModeKeys.TRAIN`?